File: Taller Nº5/python/NumerosPrimos.py
# Función para verificar si un número es primo
def es_primo(n):
    if n < 2:  # Los números menores que 2 no son primos
        return False
    for i in range(2, int(n**0.5) + 1):  # Verificamos hasta la raíz cuadrada de n
        if n % i == 0:  # Si es divisible por otro número, no es primo
            return False
    return True  # Si no encontramos divisores, es primo

# Función para sumar primos en el rango [a, b]
def suma_primos_en_rango(a, b):
    suma = 0
    for i in range(min(a, b), max(a, b) + 1):  # Recorremos desde el menor hasta el mayor
        if es_primo(i):  # Si el número es primo, lo sumamos
            suma += i
    return suma

# Solicitar al usuario los dos números
a = int(input("Ingresa el primer número: "))
b = int(input("Ingresa el segundo número: "))

# Calcular e imprimir el resultado
resultado = suma_primos_en_rango(a, b)
print(f"La suma de los números primos entre {a} y {b} es: {resultado}")

File: Taller Nº5/Java/NumerosPrimos.java
import java.util.Scanner; // Importamos Scanner para leer entrada del usuario

public class Main { // Clase principal
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in); // Creamos objeto Scanner para entrada

        System.out.print("Ingresa el primer número: ");
        int a = scanner.nextInt(); // Leemos el primer número

        System.out.print("Ingresa el segundo número: ");
        int b = scanner.nextInt(); // Leemos el segundo número

        int resultado = sumaPrimosEnRango(a, b); // Llamamos a la función para obtener la suma

        System.out.println("La suma de los números primos entre " + a + " y " + b + " es: " + resultado);
    }

    // Función para verificar si un número es primo
    public static boolean esPrimo(int n) {
        if (n < 2) {
            return false; // Los números menores a 2 no son primos
        }
        for (int i = 2; i <= Math.sqrt(n); i++) { // Recorremos hasta la raíz cuadrada
            if (n % i == 0) {
                return false; // Si tiene un divisor, no es primo
            }
        }
        return true; // Si no se encontró divisor, es primo
    }

    // Función para sumar primos en el rango [a, b]
    public static int sumaPrimosEnRango(int a, int b) {
        int suma = 0;
        int inicio = Math.min(a, b); // Determinar el valor menor
        int fin = Math.max(a, b);    // Determinar el valor mayor

        for (int i = inicio; i <= fin; i++) {
            if (esPrimo(i)) {
                suma += i; // Si el número es primo, lo sumamos
            }
        }
        return suma;
    }
}
